### DISCUSSION TITLE
Allow relationship fields to be used within listSearchableFields

### DISCUSSION THREAD

**Discussion by Stupidism** [payloadcms/payload#1570] [Roadmap]
When a relationship field is used within `listSearchableFields`, the List view will break because the search term entered will be used as a `regex` search on the `id` field of the related document. 

To properly support `relationship` field within `listSearchableFields`, we need to determine the field type for each specified `listSearchableField`,  and if it's a `relationship` or `upload`, format the field's query to search on the `useAsTitle` field instead of the relationship ID.

Original report below: 

# Bug Report

I tried to support suport for multiple fields in my article list page

```

    listSearchableFields: ['title', 'slug', 'description', 'tags'],
```

I got this error

```
{"errors":[{"message":"Document can't have $ prefix field names: $regex"}]}
```

And I inspected the network request found out it's caused by this query:

```
https://seo-payload.prod.getjerry.com/api/articles?locale=en&depth=0&draft=true&page=1&where[or][0][title][like]=than&where[or][1][slug][like]=than&where[or][2][description][like]=than&where[or][3][tags][like]=than

```

## Steps to Reproduce

1. Find a collection with `tags` relationship or anything similar. I don't think `hasMany` is required to reproduce this bug
2. mark that field as `listSearchableFields`
3. type anything to search in the list view you will see page blank
4. inspect the network request you will see the error.

**Comment by Stupidism** [comment]
Maybe add a check that the relationship type is not allowed in `listSearchableFields` ?
This is annoying because I cannot reproduct it in dev mode. It only shows no result

<img width="854" alt="image" src="https://user-images.githubusercontent.com/5801015/204978076-37bb0ea4-8dc5-45ef-9e43-8720f846eee0.png">

**Comment by jmikrut** [comment]
Yep - I think there are a few things we could do to improve this behavior.

Right now, the query that is made on your `tags` field is treating your search term as if it were an ID, and trying to use the search term `kentucky` as if it were a text field, but in reality, it's searching on the tag IDs which will break.

Instead, we could intelligently parse the field type for each `listSearchableField`, and if it's a relationship, we could format the search term used to search on the related collection field that is set to `useAsTitle` instead.

I'll mark this as a planned discussion feature request and we will get it added! We would also happily accept a PR here as well!

**Reply by DanRibbens** [reply]
It should not only use `useAsTitle`, but also use the `listSearchableField` from the relationship collection. It just needs to concat the relationship field name in front of the relationship field search keys.

**Comment by JessRynkar** [comment]
We should also allow **group** fields to be used in `listSearchableFields` - related discussion [here](https://github.com/payloadcms/payload/discussions/1570).

**Comment by DanRibbens** [comment]
A quick update on how to do this now even though the code here hasn't changed in a time.

It isn't documented but you can use subfield paths in your `listSearchableFields` to support many combinations beyond simply pointing to a top level field on the collection.

For example if you have a relationship field with name: 'categories' you might want to add `listSearchableFields: ['title', 'categories.name']` so that the name of your categories is can be used to give back the search results.

The same idea works for groups, arrays, etc. One limitation you might run into is with polymorphic relationships, since Payload doesn't currently support querying on values of a relationship across collections.

This should be better documented and in addition to this. I still think there is some more work to do. If a config specifies a relationship as the `listSearchableFields` we could automatically add the search fields of the related collection.

Say we have Posts and Categories and on `posts` the listSearchableFields has `category`, a relationship field. On categories we have listSearchableFields: ['name']. Then the sanitized result of `posts` `listSearchableFields` would end up with `['title', 'category.name']`. This way any change to the categories search fields would carry through to other collections.

**Reply by dlukes** [reply]
Thank you very much for the tip! I'd discovered the `.` trick for fields nested in arrays, but I had no idea it also worked with relationships :)

One remaining UX disadvantage is that the placeholder in the Search bar seems to ignore the dotted fields listed via `listSearchableFields`, so it actually gives the wrong hint. E.g. when all the listed fields are dotted, it says "Search by ID".
Currently, I work around this by overriding the translation for this placeholder:

```ts
export default buildConfig({
  // ...
  i18n: {
    translations: {
      en: {
        general: {
          // Normally contains a {{label}} variable, but the label is wrong when
          // using listSearchableFields with nested fields (shows ID instead).
          searchBy: 'Quick search',
        },
      },
    },
  },
  // ...
}
```

But maybe there's a better way?
